WebDriverAgentLib/Commands/FBElementCommands.m

```diff
@@ -608,38 +608,57 @@ + (NSArray *)routes
-  }
-  NSString* order = [request.arguments[@"order"] lowercaseString];
-  CGFloat offset = DEFAULT_OFFSET;
-  if (request.arguments[@"offset"]) {
-    offset = (CGFloat)[request.arguments[@"offset"] doubleValue];
-    if (offset <= 0.0 || offset > 0.5) {
-      return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"'offset' value is expected to be in range (0.0, 0.5]. '%@' was given instead", request.arguments[@"offset"]] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
+  + (id<FBResponsePayload>)handleWheelSelect:(FBRouteRequest *)request
+  {
+    NSNumber *counter = 0;
+    FBElementCache *elementCache = request.session.elementCache;
+    XCUIElement *element = [elementCache elementForUUID:(NSString *)request.parameters[@"uuid"]];
+    if ([element.lastSnapshot elementType] != XCUIElementTypePickerWheel) {
+      return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"The element is expected to be a valid Picker Wheel control. '%@' was given instead", element.wdType] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
     }
+    NSString* order = [request.arguments[@"order"] lowercaseString];
+    CGFloat offset = DEFAULT_OFFSET;
+    if (request.arguments[@"offset"]) {
+      offset = (CGFloat)[request.arguments[@"offset"] doubleValue];
+      if (offset <= 0.0 || offset > 0.5) {
+        return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"'offset' value is expected to be in range (0.0, 0.5]. '%@' was given instead", request.arguments[@"offset"]] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
+      }
+    }
+    BOOL isSuccessful = false;
+    NSError *error;
+    if (request.arguments[@"value"]) {
+      NSString *value = request.arguments[@"value"];
+      NSString* endValue = element.wdValue;
+      if ([element.wdValue isEqualToString: value]) {
+        return FBResponseWithOK();
+      }
+      do {
+        if ([order isEqualToString:@"previous"]) {
+          [element fb_selectPreviousOptionWithOffset:offset error:&error];
+        } else {
+          [element fb_selectNextOptionWithOffset:offset error:&error];
+        }
+        if ([element.wdValue isEqualToString: value]) {
+          return FBResponseWithOK();
+        }
+        counter = @([counter intValue] + 1);
+      } while (![element.wdValue isEqualToString: endValue] && counter.intValue <= 20);
+      return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"The target value cannot be found on the respective Picker Wheel"] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
```

Comment:
Try to assign arguments to intermediate values, so the whole string is not so long.
Also, it makes sense to be more specific in the error message how many retries it took and in which direction the swiping has been done